spec.html

```diff
@@ -119,6 +119,12 @@ <h1>
 		<emu-note type=editor>
 			In the case of an abrupt ~throw~ completion, the type of error to be created should match the type of the abrupt throw completion record. This could be revisited when merging into the main specification. Additionally, in the case of a ~break~ or ~continue~ completion, since those are not supported, a TypeError is expected.
 		</emu-note>
+		<p>
+			The host enviroment must not allow a program running in a ShadowRealm to observe the identity of an object from any other Realm. Similarly, the host enviroment must not allow a program running in a Realm to observe the identity of an object from a ShadowRealm.
+		</p>
+		<emu-note type=editor>
+			If an execution in a ShadowRealm R1 oblivious of host APIs can observe the identity of an object O1, a host API must not allow an execution in any other realm than R1 to also observe the identity of O1. Similarly if an execution in a realm R2 can observe the identity of an object O2, a host API must not allow execution in any other Realm than R2 that is a ShadowRealm to also observe the identity of O2. The text above imposes the callable boundary semantics only when at least one of the two realms involved is a ShadowRealm. Other Realms can continue sharing objects whose identities can be observed.
```

Comment:
I understand what this intends, but the wording threw me off because I thought identity was conceptual only. Identity-the-concept can be reified in various ways (some of which are not forbidden, e.g. symbol mapping). What’s forbidden is the \<mumble\>cross-realm sharing\<\/mumble\> of concrete, first-class language value objects. These happen to have identity, but then so do symbols, so why is “object identities” the subject rather than “objects”?

Reply:
@bathos I'm not sure I understand the distinction you are asking for. Observing an object and observing its identity seem synonymous to me, but observable identity of objects is used elsewhere such as in the definition of [liveness](https://tc39.es/ecma262/#sec-liveness). If you can explain a bit more, I'd be happy to update this.